### PR TITLE
Add sqls support

### PR DESCRIPTION
**Pull request by ray-x**
sqls : https://github.com/lighttiger2505/sqls
golang implementation of SQL lsp.

### CODE REVIEW

File: lua/lspconfig/sqls.lua [hunk 1]
```diff
@@ -0,0 +1,31 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local root_pattern = util.root_pattern(".git", "config.yml")
+
+configs.sqls = {
+  default_config = {
+    cmd = {"sqls", "-config", "$HOME/.config/sqls/config.yml"};
```
Comment (nanotee) [inline lua/lspconfig/sqls.lua:8]: Is the `-config` flag necessary here? According to the README, `sqls` reads from `$XDG_CONFIG_HOME` automatically: https://github.com/lighttiger2505/sqls#configuration-methods

**Comment by mjlbach** [comment]
@ray-x  Rebase, don't edit README.md/CONFIG.md, and address @nanotee's point and I will merge!

**Comment by mjlbach** [comment]
Also add installation instructions (if pertinent) following that of other language servers, and please squash your commits

File: lua/lspconfig/sqls.lua (hunk 1 above)
Comment (ray-x) [reply lua/lspconfig/sqls.lua]: Agree, the default config should not setup config.yml to a specific folder. It was updated in latest push

**Comment by ray-x** [comment]
Update install instructions. Branch squashed.

**Comment by mjlbach** [comment]
I think something went wrong here 😆 image

**Comment by ray-x** [comment]
Rebase on a old version. Fixed.

File: lua/lspconfig/sqls.lua [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+require'lspconfig'.sqls.setup{
+  cmd = {"path/to/command", "-config" "path/to/config.yml"};
+  ...
+}
+
+This LSP can be installed via `go get github.com/lighttiger2505/sqls` Find further instructions on manual installation of the Language Server Protocol for SQL at [lighttiger2505/sqls](https://github.com/lighttiger2505/sqls)
+
+```
+
+<br>
```
Comment (mjlbach) [inline lua/lspconfig/sqls.lua:29]: why <br> here?

File: lua/lspconfig/sqls.lua [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+```lua
+require'lspconfig'.sqls.setup{
+  cmd = {"path/to/command", "-config" "path/to/config.yml"};
+  ...
+}
+
+This LSP can be installed via `go get github.com/lighttiger2505/sqls` Find further instructions on manual installation of the Language Server Protocol for SQL at [lighttiger2505/sqls](https://github.com/lighttiger2505/sqls)
+
+```
```
Comment (mjlbach) [inline lua/lspconfig/sqls.lua:27]: I think these backticks should be up by the lua block

Comment (mjlbach) [reply lua/lspconfig/sqls.lua:27]:
```
Sqls can be installed via `go get github.com/lighttiger2505/sqls`. Instructions for compiling Sqls from the source can be found at [lighttiger2505/sqls](https://github.com/lighttiger2505/sqls).
```
```

File: lua/lspconfig/sqls.lua [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+local util = require 'lspconfig/util'
+
+local root_pattern = util.root_pattern(".git", "config.yml")
+
+configs.sqls = {
+  default_config = {
+    cmd = {"sqls"};
+    filetypes = {"sql", "mysql"};
+    root_dir = function(fname)
+      return root_pattern(fname) or vim.loop.os_homedir()
```
Comment (mjlbach) [inline lua/lspconfig/sqls.lua:11]: Should it really open in home directory? Cwd? Buf directory? (for fallback)

File: lua/lspconfig/sqls.lua (hunk 2 above)
Comment (ray-x) [reply lua/lspconfig/sqls.lua]: <br> is not needed. Could be removed

File: lua/lspconfig/sqls.lua (hunk 3 above)
Comment (ray-x) [reply lua/lspconfig/sqls.lua:24]: Thanks. Updated backticks and document

**Comment by mjlbach** [comment]
Ok, cool, after you take care of the home directory thing (if that's what it should be, totally ok to keep it), and squash I'll merge

File: lua/lspconfig/sqls.lua (hunk 4 above)
Comment (ray-x) [reply lua/lspconfig/sqls.lua]: changed to cwd/.git

File: lua/lspconfig/sqls.lua [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+local util = require 'lspconfig/util'
+
+local root_pattern = util.root_pattern(".git", "config.yml")
+
+configs.sqls = {
+  default_config = {
+    cmd = {"sqls"};
+    filetypes = {"sql", "mysql"};
+    root_dir = function(fname)
+      return root_pattern(fname) or util.root_pattern(vim.fn.getcwd(), ".git")
```
Comment (mjlbach) [inline lua/lspconfig/sqls.lua:11]:
```lua    
    root_dir = function(filename)
      return util.root_pattern(".git", "config.yml")(filename) or
             util.path.dirname(filename)
    end;
```

**Pull request merged**